src/routes/(console)/project-[project]/messaging/helper.ts

```diff
@@ -0,0 +1,43 @@
+import { sdk } from '$lib/stores/sdk';
+import { type Models } from '@appwrite.io/console';
+
+/** Stores active polling intervals for messages. */
+const messageIntervals = new Map<string, ReturnType<typeof setInterval>>();
+
+/** Checks the status of a message and stops polling if it's no longer processing. */
+function checkMessageStatus(message: Models.Message) {
```

Comment:
Any performance problems if there's a full page of processing messages? Also, what if the status doesn't change from processing?

Reply:
1. Shouldn't be too much of an overhead. Once the page is moved from, the polling stops via `onDestroy`.
2. That's a tricky one, we could add some sort of fallback in retries or a timer but I haven't seen a case where a message was stuck on `processing` indefinitely. It fails pretty quick on an issue.

Reply:
> I haven't seen a case where a message was stuck on processing indefinitely. It fails pretty quick on an issue.

Maybe they were more frequent in the past and we've fixed them by now.